.augment/rules/examples.md

```diff
@@ -0,0 +1,371 @@
+# Code Examples and Anti-Patterns
+
+This document provides comprehensive examples of correct and incorrect code patterns for the Rubrik Polaris SDK for Go.
+
+## Table of Contents
+
+1. [Documentation Examples](#documentation-examples)
+2. [Acronym Capitalization Examples](#acronym-capitalization-examples)
+3. [GraphQL Query Examples](#graphql-query-examples)
+4. [Region Type Examples](#region-type-examples)
+5. [Complete Code Examples](#complete-code-examples)
+
+## Documentation Examples
+
+### ✅ Correct: Exported Type with Documentation
+
+```go
+// CloudAccount represents an AWS cloud account in Rubrik Security Cloud.
+// It contains the account metadata and enabled features.
+type CloudAccount struct {
+    // ID is the Rubrik cloud account identifier.
+    ID uuid.UUID
+    
+    // NativeID is the AWS account ID.
+    NativeID string
+    
+    // Name is the friendly name of the cloud account.
+    Name string
+    
+    // Features contains the list of enabled features for this account.
+    Features []Feature
+}
+```
+
+### ✅ Correct: Exported Function with Documentation
+
+```go
+// AddAccount adds a new AWS cloud account to Rubrik Security Cloud.
+// It returns the cloud account ID on success or an error if the operation fails.
+//
+// Example:
+//   id, err := api.AddAccount(ctx, AccountID("123456789012"), Name("Production"))
+//   if err != nil {
+//       return err
+//   }
+func (a API) AddAccount(ctx context.Context, opts ...OptionFunc) (uuid.UUID, error) {
+    // implementation
+}
+```
+
+### ❌ Incorrect: Missing Documentation
+
+```go
+type CloudAccount struct {
+    ID       uuid.UUID
+    NativeID string
+    Name     string
+    Features []Feature
+}
+
+func (a API) AddAccount(ctx context.Context, opts ...OptionFunc) (uuid.UUID, error) {
+    // implementation
+}
+```
+
+## Acronym Capitalization Examples
+
+### ✅ Correct: Uppercase Acronyms
+
+```go
+// Struct with proper acronym capitalization
+type AWSCloudAccountConfig struct {
+    CloudAccountID uuid.UUID
+    VPCID         string
+    SubnetIDs     []string
+    IAMARN        string
+    KMSKeyID      string
+    S3BucketURL   string
+    HTTPEndpoint  string
+    JSONConfig    string
+}
+
+// Function with proper acronym capitalization
+func (a API) GetAWSAccountByID(ctx context.Context, id uuid.UUID) (*AWSAccount, error) {
+    // implementation
+}
+
+// Constants with proper acronym capitalization
+const (
+    AWSEC2Feature  = "EC2"
+    AWSRDSFeature  = "RDS"
+    AWSS3Feature   = "S3"
+    AWSEKSFeature  = "EKS"
```

Comment:
Here we would probably have a `type AWSFeature string`, so it might make sense to name these `AWSFeatureEC2`, `AWSFeatureRDS`, etc.